### PR TITLE
INTYGFV-14792: Add support for certificate AUTO_FILL_VALIDATION

### PR DESCRIPTION
**Pull request by SamuelTilly**

### CODE REVIEW

File: packages/webcert/src/store/certificate/certificateReducer.ts [hunk 1]
```diff
@@ -269,9 +271,13 @@ const certificateReducer = createReducer(getInitialState(), (builder) =>
...
 
-      state.certificate.data[action.payload].disabled = true
-      ;(state.certificate.data[action.payload].value as ValueCode).id = ''
-      ;(state.certificate.data[action.payload].value as ValueCode).code = ''
+      const question = state.certificate.data[action.payload]
+
+      question.disabled = true
+      if (question?.value?.code) {
+        ;(question.value as ValueCode).id = ''
+        ;(question.value as ValueCode).code = ''
```
Comment (HelenaEdblad) [inline packages/webcert/src/store/certificate/certificateReducer.ts:279]: What does the ; mean? Can you elaborate?

**Review by HelenaEdblad — CHANGES_REQUESTED** [review]

**Review by mww-sopra-steria — DISMISSED** [review]
We have now done an implementation in backend. Used this PR as a guide. 

While implementing we made a change in the AUTO_FILL_VALIDATION "model". Instead of having a AutoFillValidationType, we used the CertificateValueType and believe that the AutoFillValidationType could be redundant.

What do you think about this? See screenshot of json-data retrieved from backend.

image

File: packages/webcert/src/store/certificate/certificateReducer.ts (hunk 1 above)
Comment (SamuelTilly) [reply packages/webcert/src/store/certificate/certificateReducer.ts:279]: Good question, for some reason prettier want's this, I guess it's having issues with parsing instances where type assignment and variable assignment are done simultaneous. I am not entirely sure.

2022-10-07 10_56_45-● certificateReducer ts - frontend - Visual Studio Code

**Comment by SamuelTilly** [comment]
> We have now done an implementation in backend. Used this PR as a guide.
> 
> While implementing we made a change in the AUTO_FILL_VALIDATION "model". Instead of having a AutoFillValidationType, we used the CertificateValueType and believe that the AutoFillValidationType could be redundant.
> 
> What do you think about this? See screenshot of json-data retrieved from backend.
> 
> image

This makes sense! I will replace `AutoFillValidationType` with `CertificateDataValueType` for this PR as well.

**Review by HelenaEdblad — APPROVED** [review]

**Pull request merged**